File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int numCpy = numberToCheck, digitCount = 0, result = 0;

        while (numCpy > 0) {
            digitCount++;
            numCpy /= 10;
        }

        numCpy = numberToCheck;

        while (numCpy > 0){
            int digitToPow = 1;
            for (int i = 0; i < digitCount; i++) {
                digitToPow *= numCpy % 10;
            }
            result += digitToPow;
            numCpy /= 10;
        }

        return result == numberToCheck;
    }

}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

static int ipow(int x, int y)
{
    int result = 1;
    while (y > 0)
    {
        result *= x;
        y--;
    }
    return result;
}

bool is_armstrong_number(int candidate)
{
    int digits_sum = 0;
    int candidate_init_val = candidate;

    int no_digits = 0;

    while (candidate > 0)
    {
        no_digits++;
        candidate /= 10;
    }

    candidate = candidate_init_val;

    while (candidate > 0)
    {
        digits_sum += ipow((candidate % 10), no_digits);
        candidate /= 10;
    }

    return digits_sum == candidate_init_val;
}
